File: ui-api/src/main/java/io/tipsters/uiapi/domain/Competition.java
package io.tipsters.uiapi.domain;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "competitions")
public class Competition extends BaseEntity {

  private String name;

  private int rank;

  private boolean active;

  @ManyToOne
  @JoinColumn(name = "country_id")
  private Country country;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public int getRank() {
    return rank;
  }

  public void setRank(int rank) {
    this.rank = rank;
  }

  public boolean isActive() {
    return active;
  }

  public void setActive(boolean active) {
    this.active = active;
  }

  public Country getCountry() {
    return country;
  }

  public void setCountry(Country country) {
    this.country = country;
  }
}

File: web-server/src/main/kotlin/io/tipsters/domain/Competition.kt
package io.tipsters.domain

import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "competitions")
class Competition (
        var name: String = "",
        var rank: Int = 0,
        var active: Boolean = true,
        @ManyToOne @JoinColumn(name = "country_id") var country: Country) : BaseEntity() {
    constructor() : this("", 0, false, Country())
}
